ntoskrnl/cc/pin.c

```diff
@@ -585,6 +585,9 @@ CcUnpinDataForThread (
 
     if (iBcb->PinCount != 0)
     {
+        /* Assign resource thread to BCB */
+        iBcb->Lock.OwnerEntry.OwnerThread = ResourceThreadId;
```

Comment:
Oh no. You can't change the owner of a resource just because it "fixes" things. Please open a bug with a proper backtrace.

Reply:
> Please open a bug with a proper backtrace.

The log with a proper backtrace(s) seems to be already attached to that ticket: https://jira.reactos.org/secure/attachment/59031/NTFS.log.
The backtrace from the assert is:
```
Eip:
<ntoskrnl.exe:1820ee (/media/oleg/439750a4-876d-45c9-949c-6083bd424cbf/reactos/output-MinGW-i386/../sdk/lib/rtl/i386/debug_asm.S:33 (DbgBreakPoint))>
Frames:
<ntoskrnl.exe:34093 (ntoskrnl/ex/resource.c:1934 (ExReleaseResourceForThreadLite))>
<ntoskrnl.exe:762d (ntoskrnl/cc/pin.c:591 (CcUnpinDataForThread))>
<ntoskrnl.exe:764f (ntoskrnl/cc/pin.c:570 (CcUnpinData))>
<ntfs.sys:4679c>
<ntfs.sys:4a9ff>
<ntfs.sys:47ef3>
<ntoskrnl.exe:6f853 (ntoskrnl/io/iomgr/irp.c:1286 (IofCallDriver))>
<ntoskrnl.exe:63208 (ntoskrnl/io/iomgr/file.c:997 (IopParseDevice))>
<ntoskrnl.exe:110df6 (ntoskrnl/ob/obname.c:1003 (ObpLookupObjectName))>
<ntoskrnl.exe:109635 (ntoskrnl/ob/obhandle.c:2609 (ObOpenObjectByName))>
<ntoskrnl.exe:637c4 (sdk/include/crt/mingw32/intrin_x86.h:719 (IopCreateFile))>
<ntoskrnl.exe:6445c (ntoskrnl/io/iomgr/file.c:3024 (IoCreateFile))>
<ntoskrnl.exe:650e7 (ntoskrnl/io/iomgr/file.c:3770 (NtCreateFile))>
<ntoskrnl.exe:148f89 (ntoskrnl/include/internal/i386/ke.h:712 (KiSystemServiceHandler))>
<ntoskrnl.exe:3e2f (:0 (KiFastCallEntry))>
<ntdll.dll:103d9>
<shell32.dll:87f3d>
<shell32.dll:8823a>
<shell32.dll:882d0>
<shell32.dll:81cf6>--- Press q to abort, any other key to continue ---

<shell32.dll:823d5>
<shell32.dll:7a160>
<shell32.dll:74255>
<shell32.dll:74594>
<shell32.dll:cf761>
<shell32.dll:cf2bd>
<user32.dll:67429>
<user32.dll:5b398>
<user32.dll:5b7bc>
<user32.dll:5c409>
<user32.dll:3faf8>
<user32.dll:40bd7>
<uxtheme.dll:1004d>
<user32.dll:414b4>
<comctl32.dll:5522e>
<user32.dll:67429>
<user32.dll:5b398>
<user32.dll:5b7bc>
<user32.dll:5c409>
<comctl32.dll:57847>
<user32.dll:67429>--- Press q to abort, any other key to continue ---

<user32.dll:5b398>
<user32.dll:5b7bc>
<user32.dll:5bf69>
<browseui.dll:20112>
<kernel32.dll:1d0ed>
```